autolab/Autolab
# make submission summary fill up space

## Description

Previously, assignments with few problems would have a submission summary where the table only took up part of the horizontal space provided:

<img width="1197" alt="Screen Shot 2022-10-13 at 22 20 35" src="https://user-images.githubusercontent.com/25730111/195887072-ec190716-68b3-420c-95a7-28b62d0de8aa.png">

The "show all submissions" button also displayed for any amount of submissions, which was weird in the case where only a couple submissions were made, since the information in history is now exactly the same as submission summary:

<img width="1066" alt="Screen Shot 2022-10-18 at 18 19 14" src="https://user-images.githubusercontent.com/25730111/196555766-8eff6fa5-0ffd-453a-a5fc-53e4defa39d6.png">

## Change Summary

- Changed the table view to now use css that applies `display: table;` to the actual table, and `overflow-x: scroll` to the wrapper div so that the table will become scrollable if there are too many problems, but take up the space otherwise.
- Changed "Download Submission" and "View Source" into buttons so that the options are easier to click
<img width="1169" alt="Screen Shot 2022-10-18 at 18 16 48" src="https://user-images.githubusercontent.com/25730111/196555653-2dbe8c06-37b4-428f-b57c-8d121370c76a.png">


- Changed the logic behind the submissions history link so that it only displays when the total submission count exceeds the amount of submissions the summary can display. 
  - The summary table is the exact same as the history, so this change doesn't change much, just makes more logical sense.

![Screen Shot 2022-10-14 at 11 39 04](https://user-images.githubusercontent.com/25730111/195887845-d40933d9-a509-4a08-81aa-9fce433de26a.png)

Lots of problems:
<img width="1209" alt="Screen Shot 2022-10-18 at 18 15 11" src="https://user-images.githubusercontent.com/25730111/196556547-e29bd3cb-d1d4-4437-a89e-ccb92db87a0c.png">


## How Has This Been Tested?



- Submit to an assignment with one problem, see that summary takes up horizontal space
- Submit to an assignment with lots of problems, see that summary scrolls properly
- Submit to an assignment with no previous submissions, see that link to "See all x submissions" doesn't appear until after the fourth submission
- Download Submission / View Submission
- Upload plain text file, see that visually the button looks the same, but hovering over shows tooltip that indicates to download as text
## Types of changes

## Checklist:
